test/lib/expr_test_utils.hh

```diff
@@ -98,6 +98,57 @@ std::pair<evaluation_inputs, std::unique_ptr<evaluation_inputs_data>> make_evalu
+                                                                       std::string_view name) const override;
+    virtual std::vector<data_dictionary::keyspace> get_keyspaces(data_dictionary::database db) const override;
+    virtual std::vector<data_dictionary::table> get_tables(data_dictionary::database db) const override;
+    virtual std::optional<data_dictionary::table> try_find_table(data_dictionary::database db,
+                                                                 std::string_view ks,
+                                                                 std::string_view tab) const override;
+    virtual std::optional<data_dictionary::table> try_find_table(data_dictionary::database db,
+                                                                 table_id id) const override;
+    virtual const secondary_index::secondary_index_manager& get_index_manager(data_dictionary::table t) const override;
+    virtual schema_ptr get_table_schema(data_dictionary::table t) const override;
+    virtual lw_shared_ptr<data_dictionary::keyspace_metadata> get_keyspace_metadata(
+        data_dictionary::keyspace ks) const override;
+    virtual bool is_internal(data_dictionary::keyspace ks) const override;
+    virtual const locator::abstract_replication_strategy& get_replication_strategy(
+        data_dictionary::keyspace ks) const override;
+    virtual const std::vector<view_ptr>& get_table_views(data_dictionary::table t) const override;
+    virtual sstring get_available_index_name(data_dictionary::database db,
+                                             std::string_view ks_name,
+                                             std::string_view table_name,
+                                             std::optional<sstring> index_name_root) const override;
+    virtual std::set<sstring> existing_index_names(data_dictionary::database db,
+                                                   std::string_view ks_name,
+                                                   std::string_view cf_to_exclude = {}) const override;
+    virtual schema_ptr find_indexed_table(data_dictionary::database db,
+                                          std::string_view ks_name,
+                                          std::string_view index_name) const override;
+    virtual schema_ptr get_cdc_base_table(data_dictionary::database db, const schema&) const override;
+    virtual const db::config& get_config(data_dictionary::database db) const override;
+    virtual const db::extensions& get_extensions(data_dictionary::database db) const override;
+    virtual const gms::feature_service& get_features(data_dictionary::database db) const override;
+    virtual replica::database& real_database(data_dictionary::database db) const override;
+};
+
+// Creates a mock implementation of data_dictionary::database, useful in tests.
+std::pair<data_dictionary::database, std::unique_ptr<mock_database_impl>> make_data_dictionary_database(
```

Comment:
If you change this to return std::pair<..., std::unique_ptr<data_dictionary::impl>> it will still work, and allow you to move the definition out-of-line.